monkestation/code/modules/vehicles/monkey_ball.dm

```diff
@@ -0,0 +1,43 @@
+/obj/vehicle/ridden/monkey_ball
+	name = "Monkey Ball"
+	desc = "An ominously monkey-sized ball."
+	icon = 'monkestation/icons/obj/vehicles.dmi'
+	icon_state = "monkey_ball"
+	max_integrity = 100
+	layer = ABOVE_MOB_LAYER
+	movedelay = 1
+	var/rotation = 0
+	var/last_bump = 0
+
+/obj/vehicle/ridden/monkey_ball/Initialize(mapload)
+	. = ..()
+	var/datum/component/riding/riding_component = LoadComponent(/datum/component/riding)
+	riding_component.vehicle_move_delay = movedelay
+	riding_component.set_vehicle_dir_layer(NORTH, ABOVE_MOB_LAYER)
+
+/obj/vehicle/ridden/monkey_ball/Destroy()
+	if(has_buckled_mobs())
+		var/mob/living/carbon/driver = buckled_mobs[1]
+		unbuckle_mob(driver)
+		driver.throw_at(get_edge_target_turf(driver,pick(GLOB.alldirs)),rand(1, 10),rand(1, 10))
+		driver.Knockdown(12 SECONDS)
+	return ..()
+
+/obj/vehicle/ridden/monkey_ball/Moved()
+	rotation += (dir == NORTH || dir == WEST || dir == NORTHWEST) ? -30 : 30
```

Comment:
Still getting the issue where it won't rotate when moving in a purely diagonal direction.
I think the issue may be because it's doing both at the same time, which cancels out the animation.

Reply:
With the way I've done it now, there still are tiny stutters occasionally, but for the most part if you are moving diagonally then the ball turns smoothly as well